### PR TITLE
RACSignal+Operations: implement non-recursive +combineLatest:.

### PR DESCRIPTION
**Pull request by StatusReport**

### CODE REVIEW

File: ReactiveObjC/RACSignal+Operations.m [hunk 1]
```diff
@@ -451,8 +451,55 @@ - (RACSignal *)combineLatestWith:(RACSignal *)signal {
...
+		__block NSUInteger completedCount = 0;
+		// Set of signal indices that already sent at least one value.
+		__block NSMutableSet<NSNumber *> *sentValues = [NSMutableSet set];
+		// Last value that was sent by each signal.
+		__block NSMutableArray *lastValue = [NSMutableArray arrayWithCapacity:signalsCount];
+		for (NSUInteger i = 0; i < signalsCount; ++i) {
+			[lastValue addObject:[NSNull null]];
+		}
+
+		void (^sendNext)(NSUInteger) = ^(NSUInteger index) {
```
Comment (barakyoresh) [inline ReactiveObjC/RACSignal+Operations.m]: You never use `index`, why bother sending it?

File: ReactiveObjC/RACSignal+Operations.m [hunk 2]
```diff
@@ -451,8 +451,55 @@ - (RACSignal *)combineLatestWith:(RACSignal *)signal {
...
+			[subscriber sendCompleted];
+			return nil;
+		}
+
+		// Number of signals that completed.
+		__block NSUInteger completedCount = 0;
+		// Set of signal indices that already sent at least one value.
+		__block NSMutableSet<NSNumber *> *sentValues = [NSMutableSet set];
+		// Last value that was sent by each signal.
+		__block NSMutableArray *lastValue = [NSMutableArray arrayWithCapacity:signalsCount];
```
Comment (barakyoresh) [inline ReactiveObjC/RACSignal+Operations.m]: isn't `lastValues` more appropriate?

File: ReactiveObjC/RACSignal+Operations.m [hunk 3]
```diff
@@ -451,8 +451,55 @@ - (RACSignal *)combineLatestWith:(RACSignal *)signal {
...
+					[sentValues addObject:@(i)];
+					lastValue[i] = x ?: RACTupleNil.tupleNil;
+					sendNext(i);
+				}
+			} error:^(NSError *error) {
+				[subscriber sendError:error];
+			} completed:^{
+				@synchronized (disposable) {
+					++completedCount;
+					if (completedCount == signalsCount) [subscriber sendCompleted];
```
Comment (barakyoresh) [inline ReactiveObjC/RACSignal+Operations.m]: blasphemy.

File: ReactiveObjC/RACSignal+Operations.m (hunk 2 above)
Comment (StatusReport) [inline ReactiveObjC/RACSignal+Operations.m]: Indeed, changed.

File: ReactiveObjC/RACSignal+Operations.m (hunk 1 above)
Comment (StatusReport) [inline ReactiveObjC/RACSignal+Operations.m]: Removed.

File: ReactiveObjC/RACSignal+Operations.m (hunk 3 above)
Comment (StatusReport) [inline ReactiveObjC/RACSignal+Operations.m]: I believe I didn't write that - it was probably copied from somewhere else (don't remember). Anyway we're on our own now - changing :)

**Comment by StatusReport** [comment]
Done.

**Pull request merged**